src/models/PropStore.ts

```diff
@@ -0,0 +1,56 @@
+import { InjectionKey, reactive } from 'vue'
+import { isWithComponent, isWithComponents } from '@/types/createRouteOptions'
+import { ResolvedRoute } from '@/types/resolved'
+import { Route } from '@/types/route'
+import { MaybePromise } from '@/types/utilities'
+
+export const propStoreKey: InjectionKey<PropStore> = Symbol()
+
+type ComponentProps = { id: string, name: string, props?: (params: Record<string, unknown>) => MaybePromise<Record<string, unknown>> }
+
+export class PropStore {
```

Comment:
nit/thought

We've limited the use of classes. Could we rewrite this as a factory? `createPropStore` that returns `{ setProps, getProps }`. That way we can destructure at implementation.

Reply:
the only reason I opted for the class was to match your implementation of `RouteHookStore`, do you feel like the factory pattern is more valuable than being consistent?

Reply:
Uh the route hook store should probably be a factory as well... lol

Reply:
done! I actually didn't end up using destructuring since the whole "store" gets provided/injected but I still felt like the value of factory > classes is worthwhile even without the destructuring?